### ISSUE TITLE
Software citation support in JATS

### ISSUE THREAD

**Issue by mfenner** [force11/force11-sciwg#7]
Following up on https://github.com/codemeta/codemeta/issues/19, what work is needed to properly support software citation in JATS ([Journal Article Tag Suite](https://jats.nlm.nih.gov/))?

This can build on related work done for data citation in JATS. @Daniel-Mietchen and @Klortho might be able to help, and I also have [an interest in JATS](https://github.com/mfenner/pandoc-jats).

**Comment by danielskatz** [comment]
flag as architecture?

**Comment by mfenner** [comment]
JATS for R: ttp://jats4r.org/about-jats4r

**Comment by Melissa37** [comment]
https://docs.google.com/document/d/1zztrTpTHtcdS5093j9UzIHHjgQS7-EzF2uF_WzeV9G8/edit#heading=h.eeew73cotjfj

This is an initial recommendation that requires work!

**Comment by mfenner** [comment]
@Melissa37 one question: should we try to update the JATS standard (similar to what we did with data citation), or should this be a recommendation of how to use JATS with software citations, which falls in the domain of JATS4R, or both?

**Comment by Melissa37** [comment]
The task of collecting the list of elements that should be put in a software citation could fall to this group. The detail of the JATS XML tagging could then be dealt with by the JATS4R subgroup.
That could work? I could suggest to the interested parties in the JATS subgroup to join this group as the first step?

**Comment by mfenner** [comment]
That sounds like a plan. One next step - once this group has gotten off the ground - could be to finalize the document you have started. I don't think it will take long to get there.

**Comment by Melissa37** [comment]
Thanks Martin, I've emailed the potential members:

Christopher Ingle
Content & Services Manager, IOP Publishing

August (Gus) Muench
Journals Data Scientist, American Astronomical Society

**Comment by mfenner** [comment]
I have added export of Crossref and DataCite DOI metadata to JATS `element-style` citations to `bolognese` library: https://github.com/datacite/bolognese/issues/18.

**Comment by mfenner** [comment]
The content types used by Crossref are at https://api.crossref.org/types. Ideally we align the JATS `publication-type` with those.

**Comment by mfenner** [comment]
Implemented support for JATS `element-citation` in DataCite Content Negotation as part of Force2017 hackathon. Example: https://data.datacite.org/application/vnd.jats+xml/10.5438/qeg0-3gm3

**Comment by mfenner** [comment]
Another example: https://data.datacite.org/application/vnd.jats+xml/10.5063/f1m61h5x

```
<?xml version="1.0" encoding="UTF-8"?>
<element-citation publication-type="software">
  <person-group person-group-type="author">
    <name>
      <surname>Jones</surname>
      <given-names>Matthew B.</given-names>
    </name>
    <name>
      <surname>Slaughter</surname>
      <given-names>Peter</given-names>
    </name>
    <name>
      <surname>Nahf</surname>
      <given-names>Rob</given-names>
    </name>
    <name>
      <surname>Boettiger</surname>
      <given-names>Carl</given-names>
    </name>
    <name>
      <surname>Jones</surname>
      <given-names>Chris</given-names>
    </name>
    <name>
      <surname>Read</surname>
      <given-names>Jordan</given-names>
    </name>
    <name>
      <surname>Walker</surname>
      <given-names>Lauren</given-names>
    </name>
    <name>
      <surname>Hart</surname>
      <given-names>Edmund</given-names>
    </name>
    <name>
      <surname>Chamberlain</surname>
      <given-names>Scott</given-names>
    </name>
  </person-group>
  <source>dataone: R interface to the DataONE network of data repositories</source>
  <publisher-name>KNB Data Repository</publisher-name>
  <year iso-8601-date="2016">2016</year>
  <pub-id pub-id-type="doi">10.5063/f1m61h5x</pub-id>
</element-citation>
```

**Comment by michael** [comment]
Nice! Does the software type also return a url (like project website) if available?

**Comment by mfenner** [comment]
I haven't implemented URLs yet. Why would you need that if you have the DOI?

Also, I am using `source` instead of proposing `software-title`.

**Comment by Melissa37** [comment]
Humm, not sure about that as it moves away from the standard that was set by the data citation group and also misuse of the source tag?
For data citations the JATS4R recommendation is:
<data-title> / <source>. At least one of <data-title> or <source> must be present. <data-title> should hold the title of the data set. <source> should contain the name of the holding repository.

I would rather use <data-title> than source if we're proposing not to ask for software-title

**Comment by mfenner** [comment]
I don't have enough understanding of the `source` tag, but I wouldn't call it misuse. I think it is not a good idea to add `software-title` to the already existing `article-title`, `data-title` and `chapter-title`, as this might turn into an ever-growing list. Maybe it would be better to create `title`.

**Comment by michael** [comment]
JATS is not very consequent with titles. But what I found in the spec is this:

> `article-title`:  Title of the article (Use the <source> element for titles of books, reports, conference proceedings, etc.)

So guess this would also fit for softwares.

I have no preference how to tag it, as long as we have it consistent across journals that use Texture.

@mfenner regarding the url, if the DOI forwards to the software repo or software website that's fine I guess.

**Comment by Melissa37** [comment]
I agree with Martin that proliferation of "x-title" is not the best way forward and "article-title", "data-title" could be replaced with "title" rather than adding "soft-ware" title.

There is a JATS4R group dealing with citations. I'll get their Chair details and add them to this issue.

**Comment by AlastairKelly** [comment]
Hi,  I'm new to the working group, and trying to understand if further work is needed on this issue. It's not clear to at the end of this discussion. I could not find much in the documentation for the JAST4R group. Is support for software citations in JATS considered sufficient now, or is further development needed? I'm a librarian at NLM (where JATS lives) and trying to determine if this is an issue that needs to get picked up there.

**Comment by danielskatz** [comment]
This is not an immediate answer, but as we start up the task forces, specifically the [Repository Best Practice Task Force](https://github.com/force11/force11-sciwg/issues/59), the [Publisher Adoption Task Force](https://github.com/force11/force11-sciwg/issues/58), and the [Codemeta Task Force](https://github.com/force11/force11-sciwg/issues/57), it should become more clear.

**Comment by augustfly** [comment]
>  proliferation of "x-title" is not the best way forward

@Melissa37 indeed. i'm on the JATS4R subgroup for this, and I **very much want to use** `<title>` instead of following the data citation groups' recommendation to fork out `<data-title>` or any further permutation.

**Comment by mfenner** [comment]
+1 from me for consolidating on `<title>`.

**Comment by Melissa37** [comment]
Sounds good to me!
I am not sure how to depreciate data-title

M

**Comment by tedhabermann** [comment]
+1 from me for using <title> but it is not clear how you specify the type of the content. pub-type doesn't seem right.

**Comment by augustfly** [comment]
`@publication-type="software"` as an attribute to the tags `<mixed-citation>` or `<element-citation>` seems right to me and consistent with [extant JATS 1.1](https://jats.nlm.nih.gov/archiving/tag-library/1.1/attribute/publication-type.html) and the current [JATS4R data citation recs](https://jats4r.org/data-citations).

What seems wrong?

**Comment by Melissa37** [comment]
Hi @augustfly I agree with you on using ```@publication-type="software"```

**Comment by tedhabermann** [comment]
Sorry - tricked by the difference between @publication-type and @pub-type. I agree that @publication-type looks good...
